ast.go

```diff
@@ -900,26 +1003,25 @@ var (
 	astBoolFalse = []byte("0")
 )
 
-func (node RawVal) Serialize(w io.Writer) error {
-	return encodeSQLValue(w, node.Val)
+func (node RawVal) Serialize(w Writer) error {
+	return w.WriteRaw(node)
 }
 
-// encodedVal represents an already encoded value. Not exported but
-// this struct must be used with caution.
-type encodedVal struct {
+// EncodedVal represents an already encoded value. This struct must be used
+// with caution.
```

Comment:
Probably worthwhile to expand this comment now that the type is exported. Caution must be used because usage of `EncodedVal` in query construction can allow SQL injection attacks.